<h1 style="color:#800080;text-align:center;">Week 3 - Practical Worksheet</h1>
<h2 style="color:#800080;text-align:center;">Working with data and Dataframe</h2>

<h3 b> Introduction </h3>

In the following practical worksheet, you will learn how to use different packages (Numpy, Pandas, scikit-learn), import data, and basic commands of data understanding and dataframe.
If you have any questions please ask for assistance during the session and you can also email me if you have any further questions <u> alaa.mohasseb@port.ac.uk</u>.

<u>Import Libraries</u>

In a new Jupyter Notebook (practWeek31), load the NumPy, pandas, scikit-learn and pyplot libraries:

In [1]:
#Load the numpy library
import numpy as np
#Load the pandas data-analysis library
import pandas as pd
#Import datasets from scikit-learn
from sklearn import datasets

<u>Load the Dataset</u>

First, we need to check which datasets are available.

In [36]:
datasets.load_*?

A list of the different datasest should appear as follows:

<code>datasets.load_boston
datasets.load_breast_cancer
datasets.load_diabetes
datasets.load_digits
datasets.load_files
datasets.load_iris
datasets.load_lfw_pairs
datasets.load_lfw_people
datasets.load_linnerud
datasets.load_mlcomp
datasets.load_sample_image
datasets.load_sample_images
datasets.load_svmlight_file
datasets.load_svmlight_files</code

We will examine the iris dataset, which consists of measurements of three types of Iris flowers: Iris Setosa, Iris Versicolor, and Iris Virginica.

In [46]:
iris = datasets.load_iris ()

<h3>Basics of Pandas using Iris Dataset</h3>

# <u>Displya the different variables in the dataset</u>

The output should be a NumPy array

In [47]:
iris.data

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
       [4.9, 3

<u>Displaying the shape of the dataset</u>

The shape of the dataset means to print the total number of rows or entries and the total number of columns or features of that particular dataset.

In [5]:
iris.data.shape

(150, 4)

The output shows that the data has 150 rows and 4 columns.

<u>Displaying a certain row</u>

In [6]:
iris.data[0] #first row

array([5.1, 3.5, 1.4, 0.2])

In [7]:
iris.data[3]

array([4.6, 3.1, 1.5, 0.2])

In [8]:
iris.data[10]

array([5.4, 3.7, 1.5, 0.2])

<u>to print list of features (columns name)</u>

In [9]:
iris.feature_names

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

<u>Viewing the iris dataset with Pandas</u>

The panda’s data analysis library is used to view and visualize a dataset using dataframe.

<i>Note: You can view the iris dataset with Pandas, a library built on top of NumPy.</i>

In [41]:
iris_df = pd.DataFrame(iris.data, columns = iris.feature_names)

In [42]:
print (type(iris_df))

<class 'pandas.core.frame.DataFrame'>


or

In [43]:
type(iris_df)

pandas.core.frame.DataFrame

In [44]:
#print the shape of the data
print(iris_df.shape)

(569, 30)


or

In [27]:
#print the shape of the data
iris_df.shape

(1797, 64)

<u>Displaying the top rows of the dataset with their columns</u>

The function <code>head()</code> will display the top rows of the dataset, the default value of this function is 5, that is it will show top 5 rows when no argument is given to it.

In [45]:
print(iris_df.head())

   mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0        17.99         10.38          122.80     1001.0          0.11840   
1        20.57         17.77          132.90     1326.0          0.08474   
2        19.69         21.25          130.00     1203.0          0.10960   
3        11.42         20.38           77.58      386.1          0.14250   
4        20.29         14.34          135.10     1297.0          0.10030   

   mean compactness  mean concavity  mean concave points  mean symmetry  \
0           0.27760          0.3001              0.14710         0.2419   
1           0.07864          0.0869              0.07017         0.1812   
2           0.15990          0.1974              0.12790         0.2069   
3           0.28390          0.2414              0.10520         0.2597   
4           0.13280          0.1980              0.10430         0.1809   

   mean fractal dimension  ...  worst radius  worst texture  worst perimeter  \
0           

In [29]:
#display any given (number) of rows
print(iris_df.head(10))

   pixel_0_0  pixel_0_1  pixel_0_2  pixel_0_3  pixel_0_4  pixel_0_5  \
0        0.0        0.0        5.0       13.0        9.0        1.0   
1        0.0        0.0        0.0       12.0       13.0        5.0   
2        0.0        0.0        0.0        4.0       15.0       12.0   
3        0.0        0.0        7.0       15.0       13.0        1.0   
4        0.0        0.0        0.0        1.0       11.0        0.0   
5        0.0        0.0       12.0       10.0        0.0        0.0   
6        0.0        0.0        0.0       12.0       13.0        0.0   
7        0.0        0.0        7.0        8.0       13.0       16.0   
8        0.0        0.0        9.0       14.0        8.0        1.0   
9        0.0        0.0       11.0       12.0        0.0        0.0   

   pixel_0_6  pixel_0_7  pixel_1_0  pixel_1_1  ...  pixel_6_6  pixel_6_7  \
0        0.0        0.0        0.0        0.0  ...        0.0        0.0   
1        0.0        0.0        0.0        0.0  ...        0.0     

Now, print the first 3 rows using the following function, What have you noticed?

In [17]:
iris_df.head(3)

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                5.1               3.5                1.4               0.2
1                4.9               3.0                1.4               0.2
2                4.7               3.2                1.3               0.2

The first function will view the output as a simple text however, the second function will view the output as a dataframe.

<u>Displaying the last rows of the dataset with their columns</u>

The <code>tail()</code> function is used to return the last n rows.

In [18]:
iris_df.tail()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
145                6.7               3.0                5.2               2.3
146                6.3               2.5                5.0               1.9
147                6.5               3.0                5.2               2.0
148                6.2               3.4                5.4               2.3
149                5.9               3.0                5.1               1.8

<u>Displaying the number of columns and names of the columns.</u>

The <code>column()</code> function prints all the columns of the dataset in a list form.

In [19]:
iris_df.columns

Index(['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)',
       'petal width (cm)'],
      dtype='object')

<u>Display the statistical properties of the data.</u>

The <code>describe()</code> function returns description of the data in the DataFrame. If the DataFrame contains numerical data, the description contains these information for each column: count - The number of not-empty values. mean - The average (mean) value.

In [20]:
print(iris_df.describe())

       sepal length (cm)  sepal width (cm)  petal length (cm)  \
count         150.000000        150.000000         150.000000   
mean            5.843333          3.057333           3.758000   
std             0.828066          0.435866           1.765298   
min             4.300000          2.000000           1.000000   
25%             5.100000          2.800000           1.600000   
50%             5.800000          3.000000           4.350000   
75%             6.400000          3.300000           5.100000   
max             7.900000          4.400000           6.900000   

       petal width (cm)  
count        150.000000  
mean           1.199333  
std            0.762238  
min            0.100000  
25%            0.300000  
50%            1.300000  
75%            1.800000  
max            2.500000  


or

In [ ]:
#view the output as a dataframe
iris_df.describe()

In [ ]:
#print the mean
iris_df.mean()

In [ ]:
#print the max
iris_df.max()

In [ ]:
#print the median
iris_df.median()

<i>Note: you can also use the print function <code>print(iris_df.mean())</code></i>

<u>Display dataframe values</u>

Pandas <code>DataFrame.values</code> attribute to return the numpy representation of the given DataFrame.

In [ ]:
iris_df.values

<u> Display the size of the data</u>

The <code>size</code> property returns the number of elements in the DataFrame. The number of elements is the number of rows * the number of columns.

In [ ]:
iris_df.size

<u>Count the number of records in each category</u>

The <code>counts()</code> function, counts the number of not empty values for each row, or column if you specify the axis parameter as axis='columns' , and returns a Series object with the result for each row (or column).

In [ ]:
iris_df.count()

<u>Display the data type</u>

The <code>dtypes</code> function is used to find the dtypes in the DataFrame.

This returns a Series with the data type of each column. The result’s index is the original DataFrame’s columns.

Columns with mixed types are stored with the object dtype.

In [ ]:
iris_df.dtypes